### ISSUE TITLE
Calamares is not working on openSUSE

### ISSUE THREAD

**Issue by vinyanalista** [calamares/calamares#1312]
I'm not sure this is the best place to ask, but openSUSE's [`calamares.spec`](https://build.opensuse.org/package/view_file/openSUSE:Leap:15.1:Update/calamares/calamares.spec?expand=1) pointed me here:

```
Please submit bugfixes or comments via https://github.com/calamares/calamares/issues
```

Every time I try to install an openSUSE-based live image (be the official one available at [software.opensuse.org](https://software.opensuse.org/distributions/leap), be [Linux Kamarada](https://kamarada.github.io/en/download)), the installation fails right at the beginning with:

```
Installation Failed

The installer failed to create partition on disk 'ATA VBOX HARDDISK'.

==========================================================================================
Create a new partition (19.99 GiB, ext4) on ‘/dev/sda’
========================================================================================== ==========================================================================================
Job: Create new partition on device ‘/dev/sda’
========================================================================================== ==========================================================================================
Job: Create file system ‘ext4’ on partition ‘/dev/sda1’
==========================================================================================
```

![calamares](https://user-images.githubusercontent.com/1576506/73886103-b4b72200-4847-11ea-84e5-17e98bb8c45f.png)

1. Create a VirtualBox VM
2. Boot an openSUSE-based live ISO image, such as the ones mentioned above
3. Linux Kamarada already comes with Calamares, one can install it on any openSUSE-based live image (including the official one) with:

```
# zypper in calamares
```

4. Start Calamares, select to erase the disk, and try to install
5. The error happens

openSUSE Leap 15.1 provides Calamares v3.2.15.

Do you have any idea on what is going on?

Thanks!

@embar- it seems you are the one taking care of the calamares package for openSUSE. Do you have any idea?

**Comment by wxl** [comment]
Looks to be the same as #1170 and there's no clear solution. A cheap workaround is to rename `os-prober`.

**Comment by adriaandegroot** [comment]
Yes, this is the same issue. The branch fixing it is stalled on "needs testing, also a final push, but other things keep eating my attention."

**Comment by geckolinux** [comment]
Here are two more openSUSE bugs:
- https://github.com/calamares/calamares/issues/1245
- https://github.com/OSInside/kiwi/issues/1228

**Comment by embar-** [comment]
> Here are two more openSUSE bugs:
> 
>     * #1245
> 
>     * [OSInside/kiwi#1228](https://github.com/OSInside/kiwi/issues/1228)


Already fixed in [Education](https://build.opensuse.org/package/show/Education/calamares) repository and in [openSUSE Tumbleweed](https://build.opensuse.org/request/show/758989)

**Comment by embar-** [comment]
[Submit request](https://build.opensuse.org/request/show/772308) now exist for openSUSE Leap 15.1 update.

**Comment by embar-** [comment]
Thus this https://github.com/calamares/calamares/issues/1312 should be closed as it is duplicated of other bugs or fixed.

**Comment by geckolinux** [comment]
> > Here are two more openSUSE bugs:
> > ```
> > * #1245
> > 
> > * [OSInside/kiwi#1228](https://github.com/OSInside/kiwi/issues/1228)
> > ```
> 

> Already fixed in [Education](https://build.opensuse.org/package/show/Education/calamares) repository and in [openSUSE Tumbleweed](https://build.opensuse.org/request/show/758989)

Excellent, thanks!

**Comment by vinyanalista** [comment]
You know what is curious? Without changing anything on my live system, if I just open a terminal window and start Calamares as root:

```
$ su
# calamares -d
```

I don't get that error message and installation succeeds.

**Comment by kkofler** [comment]
And how do you start Calamares when it fails? Calamares currently works only if it is run as root, by design. Any menu entries running Calamares are supposed to do that for you.

**Comment by vinyanalista** [comment]
Calamares fails when it is started through `xdg-su`.

Here is my `calamares.desktop`:

https://github.com/kamarada/Linux-Kamarada-GNOME/blob/15.1-RC3-Build43.3/root/usr/share/applications/calamares.desktop

The relevant line:

```desktop
Exec=xdg-su -c /usr/bin/calamares
```

**Comment by abucodonosor** [comment]
Why would you use xdg-su? 
Also, how does your sudo setup looks like?

**Comment by embar-** [comment]
> Already fixed in [Education](https://build.opensuse.org/package/show/Education/calamares) repository and in [openSUSE Tumbleweed](https://build.opensuse.org/request/show/758989)

Yesterday came optional update for openSUSE Leap 15.1 too (to exclude /etc/dracut.conf.d/02-livecd.conf from copying)

**Comment by kkofler** [comment]
@abucodonosor:
> Why would you use xdg-su?

I guess it's an openSUSE standard to optimize desktop integration. The obvious drawback is that you don't know what tool you will actually get under the hood. I guess the one used under GNOME is incompatible with Calamares in some way.

I have found `kdesu` to work best (whereas `pkexec` eats environment variables required for Qt desktop integration), so the Fedora Calamares package uses `kdesu` everywhere.

**Comment by abucodonosor** [comment]
@kkofler 

Yes, and it doesn't really work. You may get some random su implementation or a silent failure, or the wrong one. Similar issue but with random bash -c ..; ...; commands in #1170.

**Comment by vinyanalista** [comment]
@wxl:
> Looks to be the same as #1170 and there's no clear solution. A cheap workaround is to rename `os-prober`.

It didn't work for me on Linux Kamarada (openSUSE-based).

@abucodonosor:
> Why would you use xdg-su?

The openSUSE Calamares package uses `xdg-su` as far as I remember. It used to work until recently, when the reported error message ("The installer failed to create partition...") started to show. I don't know what changed to cause that.

Searching, I found this issue, which seems to be related: calamares/calamares#980

It suggests to use `pkexec`.

On the openSUSE Leap 15.1 official live:

```
$ pkexec /usr/bin/calamares 
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
qt.qpa.screen: QXcbConnection: Could not connect to display 
Could not connect to any X display.
```

On the openSUSE Tumbleweed official live:

```
$ pkexec /usr/bin/calamares 
/usr/bin/calamares: symbol lookup error: /usr/lib64/libcalamaresui.so.3.2.15: undefined symbol: _ZN4YAML6detail9node_data12empty_scalarB5cxx11E
```

Shortly: it does not work.

@kkofler:
> I have found `kdesu` to work best (whereas `pkexec` eats environment variables required for Qt desktop integration), so the Fedora Calamares package uses `kdesu` everywhere.

In fact, I tried that on the Linux Kamarada live image:

```
$ kdesu -c /usr/bin/calamares
```

And it seems to work.

I'm going to release a new RC with that workaround.

Thanks everyone!

**Comment by embar-** [comment]
> On the openSUSE Tumbleweed official live:
> 
> ```
> $ pkexec /usr/bin/calamares 
> /usr/bin/calamares: symbol lookup error: /usr/lib64/libcalamaresui.so.3.2.15: undefined symbol: _ZN4YAML6detail9node_data12empty_scalarB5cxx11E
> ```

This is because of updated **yaml-cpp** (i.e. **libyaml-cpp0_6**):
Seems a library changed ABI without bumping it's so-version...
https://stackoverflow.com/questions/52713571/mongod-symbol-lookup-error-mongod-undefined-symbol-zn4yaml6detail9node-data
https://forums.opensuse.org/showthread.php/538693-Blender-failed-to-launch-after-update-symbol-lookup-error-undefine-symbol
https://bugzilla.opensuse.org/show_bug.cgi?id=1160171
https://bugzilla.redhat.com/show_bug.cgi?id=1773357
https://bugzilla.redhat.com/show_bug.cgi?id=1774164
https://bbs.archlinux.org/viewtopic.php?id=248014
https://github.com/jbeder/yaml-cpp/issues/484

**Comment by kkofler** [comment]
FYI, you don't need the `-c` with `kdesu`.

And if you want to see the debugging output on the terminal, use `kdesu -t`. (This is not the default because it prevents using the background daemon `kdesud` to spawn the application and, as a result, means you have to enter the password each time, whereas `kdesud` can remember it for a few minutes.)

**Comment by embar-** [comment]
openSUSE Tumbleweed now switched into `kdesu`, see [sr#775177](https://build.opensuse.org/request/show/775177). `-c` is valid parameter for `kdesu`, though not mandatory, see [man page](https://linux.die.net/man/1/kdesu).

**Comment by kkofler** [comment]
I have not claimed that `-c` is not valid (it actually is, indeed, and I know that), just that it is not needed. :-)

**Comment by embar-** [comment]
Later openSUSE Leap 15.1 may also receive update to use `kdesu`, see [sr#775249](https://build.opensuse.org/request/show/775249)

**Comment by adriaandegroot** [comment]
All the discussion about how-do-you-start-Calamares is hardly relevant here. What **is** relevant is that os-prober was grabbing a disk and hanging around too long. I just installed Kamarada (downloaded from the link at the start of this issue) with Calamares git master (which will be 3.2.19 in a matter of hours) with no issues. That is with the merged-in kpmcore-manager branch, which tries harder to avoid races with sfdisk.

So I'm closing this as "fixed in Calamares 3.2.19, please re-spin".

**Comment by vinyanalista** [comment]
I can confirm that invoking Calamares with `kdesu` instead of `xdg-su` is a cheap workaround and works. I closed kamarada/Linux-Kamarada-GNOME#57.

Calamares is working beautifully on [Linux Kamarada 15.1 Final](https://kamarada.github.io/en/download).

Thanks!

Just out of curiosity, did you compile Calamares git master using Linux Kamarada itself? Or you copied the Calamares binaries compiled on another computer?

I remember I tried to compile Calamares v3.2.19 on [Open Build Service](https://build.opensuse.org/) and I got an error, something related to a class/method that was not implemented (missing on the Qt version packaged with openSUSE Leap 15.1)...
